src/util/rng.h

```diff
@@ -0,0 +1,64 @@
+/*********************                                                        */
+/*! \file rng.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Aina Niemetz
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2017 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief A Random Number Generator.
+ **
+ ** A random number generator, implements the xorshift* generator
+ ** (see S. Vigna, An experimental exploration of Marsaglia's xorshift
+ ** generators, scrambled. ACM Trans. Math. Softw. 42(4): 30:1-30:23, 2016).
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef __CVC4__RNG_H
+#define __CVC4__RNG_H
+
+namespace CVC4 {
+
+class RNG
+{
+ public:
+  RNG(RNG const&) = delete;
```

Comment:
I think `const RNG&` is a more common order. We should probably pick an order in the guidelines and stick with it.